### PR TITLE
[boost-locale] android build

### PR DESCRIPTION
**Pull request by xandox**
building boost-locale for android

- Which triplets are supported/not supported? mostly android 
- Have you updated the CI baseline? no

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)? yes

### CODE REVIEW

**Comment by PhoebeHui** [comment]
The gdal:x86-windows failed due to known issue https://github.com/microsoft/vcpkg/issues/15324. it doesn't block this PR.

**Review by PhoebeHui — APPROVED** [review]
LGTM, thanks for the PR!

**Comment by PhoebeHui** [comment]
@xandox, could you resolve the conflicts?

**Comment by xandox** [comment]
@PhoebeHui can you review and rerun checks, fails look like not related to this changes.

**Comment by PhoebeHui** [comment]
@xandox, please ignore the CI failures, I can't repro the issue locally, I have rerun the CI test.

File: ports/boost-locale/b2-options.cmake.in [hunk 1]
```diff
@@ -12,6 +12,23 @@ if(@VCPKG_TARGET_IS_WINDOWS@)
...
+    get_filename_component(LIBICONV_DIR "${LIBICONV_DIR}" DIRECTORY)
+
+    list(APPEND B2_OPTIONS
+        boost.locale.iconv=on
+        boost.locale.posix=off
+        /boost/locale//boost_locale
+        boost.locale.icu=${BOOST_LOCALE_ICU_FEATURE}
+        -sICONV_PATH=${LIBICONV_DIR}
+    )
+    if(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Emscripten" OR VCPKG_TARGET_ARCHITECTURE STREQUAL "wasm32")
```
Comment (dan-shaw) [inline ports/boost-locale/b2-options.cmake.in:27]: Can this logic be removed? I don't think it applies to Android.

Comment (xandox) [reply ports/boost-locale/b2-options.cmake.in:27]: Posix should be off

Comment (xandox) [reply ports/boost-locale/b2-options.cmake.in:27]: Also directory of libiconv is different

Comment (xandox) [reply ports/boost-locale/b2-options.cmake.in]: fixed

**Review by PhoebeHui — APPROVED** [review]

**Review by dan-shaw — APPROVED** [review]

**Comment by dan-shaw** [comment]
@xandox @PhoebeHui Can you resolve the conflict? Otherwise LGTM.

**Comment by xandox** [comment]
@PhoebeHui thank you

**Pull request merged**